java-client/barrage/src/main/java/io/deephaven/client/impl/BarrageSnapshotImpl.java

```diff
@@ -230,24 +230,17 @@ private void handleDisconnect() {
             return;
         }
 
-        resultTable.sealTable(() -> {
-            completed = true;
-            signalCompletion();
-        }, () -> {
-            exceptionWhileCompleting = new Exception();
-            signalCompletion();
-        });
+        completed = true;
+        signalCompletion();
         cleanup();
     }
 
-    private void signalCompletion() {
+    private synchronized void signalCompletion() {
         if (completedCondition != null) {
             resultTable.getUpdateGraph().requestSignal(completedCondition);
-        } else {
-            synchronized (BarrageSnapshotImpl.this) {
-                BarrageSnapshotImpl.this.notifyAll();
-            }
         }
+
+        BarrageSnapshotImpl.this.notifyAll();
```

Comment:
Threw me off for a minute... you're synchronized on `this`, but acting like you're signaling on a different object.
```suggestion
        notifyAll();
```